src/pool/srv_pool.c

```diff
@@ -1256,45 +1251,10 @@ read_db_for_stepping_up(struct pool_svc *svc, struct pool_buf **map_buf,
 		}
 		goto out_lock;
 	}
-	if (!version_exists) {
-		/* This DB is not new and uses a layout that lacks a version. */
-		ds_notify_ras_eventf(RAS_POOL_DF_INCOMPAT, RAS_TYPE_INFO,
-				     RAS_SEV_ERROR, NULL /* hwid */,
-				     NULL /* rank */, NULL /* inc */,
-				     NULL /* jobid */,
-				     &svc->ps_uuid, NULL /* cont */,
-				     NULL /* objid */, NULL /* ctlop */,
-				     NULL /* data */,
-				     "incompatible layout version");
-		rc = -DER_DF_INCOMPT;
-		goto out_lock;
-	}
-
-	d_iov_set(&value, &global_version, sizeof(global_version));
-	rc = rdb_tx_lookup(&tx, &svc->ps_root, &ds_pool_prop_global_version, &value);
-	if (rc == -DER_NONEXIST)
-		global_version = 0;
-	else if (rc)
-		goto out_lock;
 
-	/**
-	 * downgrading the DAOS software of an upgraded pool report
-	 * a proper RAS error.
-	 */
-	if (global_version > DS_POOL_GLOBAL_VERSION) {
-		ds_notify_ras_eventf(RAS_POOL_DF_INCOMPAT, RAS_TYPE_INFO,
-				     RAS_SEV_ERROR, NULL /* hwid */,
-				     NULL /* rank */, NULL /* inc */,
-				     NULL /* jobid */,
-				     &svc->ps_uuid, NULL /* cont */,
-				     NULL /* objid */, NULL /* ctlop */,
-				     NULL /* data */,
-				     "incompatible layout version: %u larger than "
-				     "%u", global_version,
-				     DS_POOL_GLOBAL_VERSION);
-		rc = -DER_DF_INCOMPT;
-		goto out_lock;
-	}
+	if (!version_exists)
+		/* This could also be a 1.x pool, which we assume nobody cares. */
+		D_DEBUG(DB_MD, DF_UUID": assuming 2.0\n", DP_UUID(svc->ps_uuid));
```

Comment:
What would be the behavior in this case if we open a 1.x pool?  It's not so much that we don't care because there is no compatibility but at the same time, we support a 2.0 pool so what is the expected behavior if one attempts to open a  non-2.0 pool?

I think the VOS layout is incompatible so maybe it's moot?

Reply:
Original codes will reject 1.x pool, i could add codes to check if ds_pool_prop_version existed to differ this is 2.0 pool or 1.x pool and reject if pool is 1.x.

But do we really care 1.x now?

Reply:
I am not sure whether there is 1.x user to directly upgrade to 2.x system. But I am sure that 1.x vos layout is much different from 2.x. Here, even if we did not reject 1.x device, then the pool (and container and related DTX tables) also can NOT be properly loaded, either some error or some unexpected corruption.